### ISSUE TITLE
training does not converge

### ISSUE THREAD

**Issue by LevinJ** [LevinJ/SSD_tensorflow_VOC#1]
With VGG16 weights trained from ImageNET, train SSD on VOC 07 trainval dataset, the training does not converge on location loss and cross entropy loss of positive bounding box, though training on cross entropy loss of negative bounding box does seem to converge, as seen in below loss chart.
![screenshot from 2017-04-25 06-42-56](https://cloud.githubusercontent.com/assets/1629014/25414672/836df9ce-2a65-11e7-9f8f-f60010b41076.png)

**Comment by LevinJ** [comment]
After below changes, 

1) remove patch sampling and color distortion in data augmentation pipeline
2) improve the smooth L1 regularization for location loss implementation by completely following the fast rcnn paper

mAP on VOC 07trainval is 0.8, and loss chart is as below,
![screenshot from 2017-04-26 08-22-20](https://cloud.githubusercontent.com/assets/1629014/25414775/2726fc32-2a66-11e7-92cd-0ed6306ce331.png)

Hence , the root cause should probably be that there are too many negative samples in one training batch after the patch sampling. and there is unreasonable imbalance between positive and negative bounding box during the training.

Now the new problem is that the model overfit, will add in voc 12 train_val (and probably new patch sampling) and see how the training goes.

**Comment by CangHaiQingYue** [comment]
lihai!!
